services/brig/src/Brig/Effects/GalleyProvider/RPC.hs

```diff
@@ -481,3 +485,37 @@ getTeamExposeInvitationURLsToTeamAdmin tid = do
     req =
       paths ["i", "teams", toByteString' tid, "features", featureNameBS @ExposeInvitationURLsToTeamAdminConfig]
         . expect2xx
+
+checkMLSOne2OneEstablished ::
+  ( Member (Error ParseException) r,
+    Member (ServiceRPC 'Galley) r,
+    Member TinyLog r
+  ) =>
+  Local UserId ->
+  Qualified UserId ->
+  Sem r Bool
+checkMLSOne2OneEstablished self (Qualified other otherDomain) = do
+  debug $ remote "galley" . msg (val "Get the MLS one-to-one conversation")
+  response <- ServiceRPC.request @'Galley GET req
+  case HTTP.statusCode (HTTP.responseStatus response) of
+    403 -> pure False
+    400 -> pure False
+    _ {- 200 is assumed -} -> do
```

Comment:
Should we `throw` for other non-200 values?

Reply:
I think this did the job the way it is. I'm reworking this in a PR for unblocking a user, and there I take a finer grained approach.